Core/src/main/java/com/plotsquared/core/plot/Plot.java

```diff
@@ -1244,6 +1244,32 @@ public int addRunning() {
         return value;
     }
 
+    /**
+     * Sets the plot into merging process
+     */
+    public void setMerging() {
+        for (Plot plot : this.getConnectedPlots()) {
+            plot.setMeta("merging", true);
+        }
+    }
+
+    /**
+     * @return the current state of merging process
+     */
+    public Boolean isMerging() {
+        Boolean value = (Boolean) this.getMeta("merging");
+        return value != null && value;
```

Comment:
Should be enough to just return `this.getMeta("merging") != null` I guess?

Reply:
Maybe 😄

Reply:
Do we not have a hasMeta ?

Reply:
Nope

Reply:
I guess that Meta-System could use some refactoring either way. A hasMetadata extension would be great, as well updating the backing Map to an EnumMap and have enum constants for each metadata key. I guess that would benefit for maintainability and documentation purposes.

Reply:
That would be great. Lets coordinate that together I will help you out.